### PR TITLE
Bug 1713673: toggle ssh parameters if ssh-auth source secret includes a known_host…

### PR DESCRIPTION
**Pull request by gabemontero**
…s setting

manually cherr-picked the 3.11 comment in https://github.com/openshift/origin/pull/23782 given the issues there with the e2e-gcp-crio job

/assign @adambkaplan 
@openshift/openshift-team-developer-experience FYI

### CODE REVIEW

**Comment by gabemontero** [comment]
@adambkaplan is not in the 3.10 owners file

/assign @bparees

**Comment by gabemontero** [comment]
unit test failures are unrelated to this change ... this change's passed:

```
=== RUN   TestCACertHandles
--- PASS: TestCACertHandles (0.00s)
=== RUN   TestCACertSetup
--- PASS: TestCACertSetup (0.00s)
=== RUN   TestCACertSetupNoSSL
--- PASS: TestCACertSetupNoSSL (0.00s)
=== RUN   TestContextSet
--- PASS: TestContextSet (0.00s)
=== RUN   TestContextSetExisting
--- PASS: TestContextSetExisting (0.00s)
=== RUN   TestSetOverrideURL
--- PASS: TestSetOverrideURL (0.00s)
=== RUN   TestSetOverrideURLExisting
--- PASS: TestSetOverrideURLExisting (0.00s)
=== RUN   TestGitConfigHandles
--- PASS: TestGitConfigHandles (0.00s)
=== RUN   TestGitConfigSetup
--- PASS: TestGitConfigSetup (0.00s)
=== RUN   TestPasswordHandles
--- PASS: TestPasswordHandles (0.00s)
=== RUN   TestPassword
--- PASS: TestPassword (0.00s)
=== RUN   TestPresent
--- PASS: TestPresent (0.00s)
=== RUN   TestSetup
--- PASS: TestSetup (0.00s)
=== RUN   TestSSHPrivateKeyHandles
--- PASS: TestSSHPrivateKeyHandles (0.00s)
=== RUN   TestSSHPrivateKeySetup
--- PASS: TestSSHPrivateKeySetup (0.00s)
=== RUN   TestSSHPrivateKeyWithKnownHostsSetup
--- PASS: TestSSHPrivateKeyWithKnownHostsSetup (0.00s)
PASS
coverage: 61.2% of statements
ok  	github.com/openshift/origin/pkg/build/builder/cmd/scmauth	1.129s	coverage: 61.2% of statements
```

Hmmmmmmmmmmmmm

**Comment by adambkaplan** [comment]
@gabemontero looks like we missed a ruby:2.2 image:

```
2019-09-23T21:10:36.56588403Z F0923 21:10:36.564471       1 helpers.go:119] error: build error: non-zero (13) exit code from centos/ruby-22-centos7
```

**Comment by stevekuznetsov** [comment]
@adambkaplan can I get some context on that? What is failing?

**Comment by adambkaplan** [comment]
@stevekuznetsov context: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/23844/test_pull_request_origin_extended_conformance_gce_310/405/

**Comment by stevekuznetsov** [comment]
@dobbymoodge i think this has to do with the GCS token changes you were making?

**Comment by dobbymoodge** [comment]
@stevekuznetsov It shouldn't. All I did was issue a new key for an existing serviceaccount. The SA privileges shouldn't have changed. It looks like `gsutil` might be running without any auth creds at all?

**Comment by gabemontero** [comment]
@adambkaplan fyi had to redo the 2.2 to 2.5 change ... only the test that access master branch of ruby-hello-world needed 2.5 ... the rest of the 3.10 tests needed to stay on 2.2

**Comment by gabemontero** [comment]
@stevekuznetsov @dobbymoodge fwiw the `Anonymous caller does not have storage.objects.create access` is happening consistently

**Comment by gabemontero** [comment]
@adambkaplan on the lates extended_builds run, all the ruby builds passed, except for the statusfail_assemblescript test case, which is expected to fail

However, we hit our old friend, the atomic docker regression for sym links when we inject secrets:

```
2019-09-24T21:45:40.87394897Z E0924 21:45:40.872896       1 injection.go:140] Error occurred during injecting "/var/run/secrets/openshift.io/build/testsecret" to "/tmp": Error response from daemon: Error processing tar file(exit status 1): invalid symlink "/tmp/..data" -> "..2019_09_24_21_45_24.093885741"
```

Unless the testplatform guys can revert the docker version used in the AMI's for 3.10, we may be stuck here / in green button mode.

Perhaps we wait QEs verifications of the 3.9 PR that just merged, and/or the 3.11/4.1 PRs when they merge, and then see where we are.

@bparees ^^

**Comment by bparees** [comment]
> Perhaps we wait QEs verifications of the 3.9 PR that just merged,

any idea why the 3.9 PR didn't hit the docker issue?

**Comment by gabemontero** [comment]
> > Perhaps we wait QEs verifications of the 3.9 PR that just merged,
> 
> any idea why the 3.9 PR didn't hit the docker issue?

Can't run extended_builds on 3.9 any more.  Only conformance.  So it did not include the secret injection test.

**Comment by adambkaplan** [comment]
@gabemontero `prow/unit` has been failing consistently with this branch.

We may need to raise this with the build cop.

**Comment by gabemontero** [comment]
We've actually had 1 pass out of 5 @adambkaplan  ... all failures unrelated in the affected code paths.

Definitely a hurdle for us, but not *totally* broke :-)

Where we are *definitely* stuck on extended_builds per my analysis above... prioritizing that vs. non-related unit test failures at first blush seems to address the bigger evil, though sorting it all out, including the gcp item, vs. the green button, are all in the mix for sure.

**Comment by adambkaplan** [comment]
Looping in @jottofar from the new Patch team.

**Comment by gabemontero** [comment]
ok @adambkaplan @bparees per discussion in https://coreos.slack.com/archives/CBN38N3MW/p1569522460139600 I've disabled the build secret injection tests in 3.10 with a new commit here

File: test/extended/builds/secrets.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ var _ = g.Describe("[Feature:Builds][Slow] can use build secrets", func() {
 		g.Describe("build with secrets", func() {
 
 			g.It("should contain secrets during the source strategy build", func() {
+				g.Skip("skipping on 3.10 because the 3.10 AMIs have the bad docker version")
```
Comment (adambkaplan) [inline test/extended/builds/secrets.go]: @gabemontero sounds like we need openshift-specific BZs for the docker patch in RHEL 7?

Comment (gabemontero) [reply test/extended/builds/secrets.go]: yeah @bparees and I discussed this in a devex slack thread a couple of days ago ... I'll clone the existing rhel7 bz and put it once the RHEL bz ships 

@bparees did not want to burden QE with going after pre-release RHEL patches

**Comment by gabemontero** [comment]
https://github.com/openshift/release/pull/5198 did not merge before the latest unit test run started

will retest that if need be after the current one finishes

**Comment by gabemontero** [comment]
ah ... unit test passed this time @adambkaplan ...either got lucky again, or the openshift/release change got through in time

**Comment by gabemontero** [comment]
ok @adambkaplan we now have passing unit and extended_builds runs for this 3.10 PR.

Just need to sort out what is up with this old jenkins launched gcp job for 3.10.

**Comment by gabemontero** [comment]
@dobbymoodge ^^

**Comment by gabemontero** [comment]
ok we are now past install / setup issues, and on to flakes ... thanks @dobbymoodge 

one conformance haproxy test failed in the gcp test ... everything else passed

@adambkaplan .... how abput you put a lgtm on this in case we get lucky next time 

/retest

**Pull request merged**